include/zephyr/net/mdio.h

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Definitions for IEEE 802.3 management interface
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_MDIO_H_
+#define ZEPHYR_INCLUDE_NET_MDIO_H_
+
+/**
+ * @brief Definitions for IEEE 802.3 management interface
+ * @defgroup ethernet_mdio IEEE 802.3 management interface
+ * @ingroup ethernet
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** MDIO transaction operation code */
+enum mdio_opcode {
+	/** IEEE 802.3 22.2.4.5.4 write operation */
+	MDIO_OP_C22_WRITE = 1,
+
+	/** IEEE 802.3 22.2.4.5.4 read operation */
+	MDIO_OP_C22_READ = 2,
+
+	/** IEEE 802.3 45.3.4 address operation */
+	MDIO_OP_C45_ADDRESS = 0,
+
+	/** IEEE 802.3 45.3.4 write operation */
+	MDIO_OP_C45_WRITE = 1,
+
+	/** IEEE 802.3 45.3.4 post-read-increment-address operation */
+	MDIO_OP_C45_READ_INC = 2,
+
+	/** IEEE 802.3 45.3.4 read operation */
+	MDIO_OP_C45_READ = 3
+};
+
+/* MDIO Manageable Device addresses */
+/** Physical Medium Attachment / Physical Medium Dependent */
+#define MDIO_MMD_PMAPMD			1
+/** WAN Interface Sublayer */
+#define MDIO_MMD_WIS			2
+/** Physical Coding Sublayer */
+#define MDIO_MMD_PCS			3
+/** PHY Extender Sublayer */
+#define MDIO_MMD_PHYXS			4
+/** DTE Extender Sublayer */
+#define MDIO_MMD_DTEXS			5
+/** Transmission Convergence */
+#define MDIO_MMD_TC			6
+/** Auto-negotiation */
+#define MDIO_MMD_AN			7
+/** Separated PMA (1) */
+#define MDIO_MMD_SEPARATED_PMA1		8
+/** Separated PMA (2) */
+#define MDIO_MMD_SEPARATED_PMA2		9
+/** Separated PMA (3) */
+#define MDIO_MMD_SEPARATED_PMA3		10
+/** Separated PMA (4) */
+#define MDIO_MMD_SEPARATED_PMA4		11
+/** Clause 22 extension */
+#define MDIO_MMD_C22EXT			29
+/** Vendor Specific 1 */
+#define MDIO_MMD_VENDOR_SPECIFIC1	30
+/** Vendor Specific 2 */
+#define MDIO_MMD_VENDOR_SPECIFIC2	31
```

Comment:
```suggestion
#define MDIO_MMD_PMAPMD			0x01U
/** WAN Interface Sublayer */
#define MDIO_MMD_WIS			0x02U
/** Physical Coding Sublayer */
#define MDIO_MMD_PCS			0x03U
/** PHY Extender Sublayer */
#define MDIO_MMD_PHYXS			0x04U
/** DTE Extender Sublayer */
#define MDIO_MMD_DTEXS			0x05U
/** Transmission Convergence */
#define MDIO_MMD_TC			0x06U
/** Auto-negotiation */
#define MDIO_MMD_AN			0x07U
/** Separated PMA (1) */
#define MDIO_MMD_SEPARATED_PMA1		0x08U
/** Separated PMA (2) */
#define MDIO_MMD_SEPARATED_PMA2		0x09U
/** Separated PMA (3) */
#define MDIO_MMD_SEPARATED_PMA3		0x0AU
/** Separated PMA (4) */
#define MDIO_MMD_SEPARATED_PMA4		0x0BU
/** Clause 22 extension */
#define MDIO_MMD_C22EXT			0x1DU
/** Vendor Specific 1 */
#define MDIO_MMD_VENDOR_SPECIFIC1	0x1EU
/** Vendor Specific 2 */
#define MDIO_MMD_VENDOR_SPECIFIC2	0x1FU
```
I see that it is taken from [linux kernel mdio.h](https://docs.huihoo.com/doxygen/linux/kernel/3.7/uapi_2linux_2mdio_8h_source.html).

However, using hex helps here when navigating reference manuals of specific ICs.

Many defines are not used in Zephyr drivers. Should they be kept?

Reply:
> Many defines are not used in Zephyr drivers. Should they be kept?

I tried to add the minimal currently used, but on the other hand I didn't want to partially define the generic MMD devices and registers. In this way, we set the base definitions for future drivers to use them and can be extend as they see need.